clients/drcachesim/reader/zipfile_file_reader.cpp

```diff
@@ -130,4 +131,100 @@ file_reader_t<zipfile_reader_t>::is_complete()
     return false;
 }
 
-// TODO i#5538: Implement seeking via unzLocateFile.
+template <>
+bool
+file_reader_t<zipfile_reader_t>::skip_thread_instructions(size_t thread_index,
+                                                          uint64_t instruction_count,
+                                                          OUT bool *eof)
+{
+    if (instruction_count == 0)
+        return true;
+    VPRINT(this, 2, "Thread #%zd skipping %" PRIi64 " instrs\n", thread_index,
+           instruction_count);
+    trace_entry_t timestamp = {};
+    trace_entry_t cpu = {};
+    const memref_t &memref = **this;
+    if (memref.marker.type == TRACE_TYPE_MARKER &&
+        memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP) {
+        timestamp = entry_copy_;
+    }
+    zipfile_reader_t *zipfile = &input_files_[thread_index];
+    // We assume our unzGoToNextFile loop is plenty performant and we don't need to
+    // know the chunk names to use with a single unzLocateFile.
+    uint64_t stop_count_ = cur_instr_count_ + instruction_count + 1;
+    VPRINT(this, 2,
+           "stop=%" PRIi64 " cur=%" PRIi64 " chunk=%" PRIi64 " est=%" PRIi64 "\n",
+           stop_count_, cur_instr_count_, chunk_instr_count_,
+           cur_instr_count_ +
+               (chunk_instr_count_ - (cur_instr_count_ % chunk_instr_count_)));
+    while (cur_instr_count_ +
```

Comment:
(Optional). An alternative is
```
(cur_instr_count_/chunk_instr_count_ + 1)*chunk_instr_count_ < stop_count_
```
Seemed a bit simpler to me.

Reply:
Hmm, could see it either way: leaning toward current b/c it reads as adding to the current.